### PR TITLE
Script to re prepare submissions for export

### PR DESCRIPTION
**Pull request by claire-lovisa**
:warning: TODO: bump prepare-submissions-for-export service once the fix is approved, merged and tagged

# Issue

Some submissions have not been flagged for export even though they should have, see [here](https://github.com/lblod/prepare-submissions-for-export-service/pull/15) for the details. We need to feed them again to the `prepare-submissions-for-export-service` for them to be flagged properly, once the fix is merged.

# Proposed solution

This is a half-automated solution. Here are the steps.

1. Get the form data URIs to re-submit

```
select distinct ?formData where {
  ?submission a <http://rdf.myexperiment.org/ontologies/base/Submission> ;
    <http://www.w3.org/ns/prov#generated> ?formData ;
    <http://purl.org/dc/terms/created> ?created .

  ?formData <http://mu.semte.ch/vocabularies/ext/formSubmissionStatus> <http://lblod.data.gift/concepts/9bd8d86d-bb10-4456-a84e-91e9507c374c> ;
    <http://mu.semte.ch/vocabularies/ext/decisionType> ?firstDecisionType ;
    <http://mu.semte.ch/vocabularies/ext/decisionType> ?secondDecisionType .

  ?firstDecisionType skos:prefLabel ?firstDecisionTypeLabel .
  ?secondDecisionType skos:prefLabel ?secondDecisionTypeLabel .

  FILTER (?firstDecisionType > ?secondDecisionType)

  FILTER NOT EXISTS {
    ?submission <http://schema.org/publication> ?flag.
  }
}
```

2. Copy them in the `form-data-uris.txt` file. I already put in there the URIs found on the current state of the database, but new URIs might land in this wrong state too before we deploy it.

3. Run the script: ` ./run.sh <ip-of-prepare-submissions-for-export-container`

# Expected result

You can use the following query to monitor the submissions with multiple decision types that have not been flagged:
```
select distinct ?submission ?created ?firstDecisionTypeLabel ?secondDecisionTypeLabel where {
  ?submission a <http://rdf.myexperiment.org/ontologies/base/Submission> ;
    <http://www.w3.org/ns/prov#generated> ?formData ;
    <http://purl.org/dc/terms/created> ?created .

  ?formData <http://mu.semte.ch/vocabularies/ext/formSubmissionStatus> <http://lblod.data.gift/concepts/9bd8d86d-bb10-4456-a84e-91e9507c374c> ;
    <http://mu.semte.ch/vocabularies/ext/decisionType> ?firstDecisionType ;
    <http://mu.semte.ch/vocabularies/ext/decisionType> ?secondDecisionType .

  ?firstDecisionType skos:prefLabel ?firstDecisionTypeLabel .
  ?secondDecisionType skos:prefLabel ?secondDecisionTypeLabel .

  FILTER (?firstDecisionType > ?secondDecisionType)

  FILTER NOT EXISTS {
    ?submission <http://schema.org/publication> ?flag.
  }
}
order by ?submission
```

During my tests locally, on a copy of prod data, I went from 264 un-flagged submissions to 30 after running the script.

# Relates to
- The story  [DL-6182]
- The fix in `prepare-submissions-for-export-service` [here](https://github.com/lblod/prepare-submissions-for-export-service/pull/15)

### CODE REVIEW

File: scripts/send-submissions-to-prepare-for-export/run.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+if [ "$#" -ne 1 ]; then
+  echo "The number of parameters is not correct. You need to provide the following: ./run.sh ipContainerPrepareSubmissionsForExport"
+  exit 1
+fi
+
+CONTAINER_IP=$1
+
+while IFS="" read -r l || [ -n "$l" ]
+do
+  curl --location --request POST $CONTAINER_IP/delta \
```
Comment (Riadabd) [inline scripts/send-submissions-to-prepare-for-export/run.sh:12]: Would it be better to have a one-liner to directly fetch the container IP? This way we don't have to manually do it and pass it to the script:
```suggestion
  curl --location --request POST "$(docker inspect "$(docker compose ps -q prepare-submissions-for-export)" --format '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}')"/delta \
```

File: scripts/send-submissions-to-prepare-for-export/run.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  --header 'Content-Type: application/vnd.api+json' \
+  --data-raw ' [
+    {
+      "deletes": [],
+      "inserts": [
+        {
+          "subject": { "type": "uri", "value": "'$l'" },
```
Comment (Riadabd) [inline scripts/send-submissions-to-prepare-for-export/run.sh:19]: This was more a suggestion from the VSCode shellcheck extension. It refers to [this potential issue](https://www.shellcheck.net/wiki/SC2086) that may occur if double-quotes are not wrapped around variables.

```suggestion
          "subject": { "type": "uri", "value": "'"$l"'" },
```

File: scripts/send-submissions-to-prepare-for-export/run.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  --data-raw ' [
+    {
+      "deletes": [],
+      "inserts": [
+        {
+          "subject": { "type": "uri", "value": "'"$l"'" },
+          "predicate": { "type": "uri", "value": "http://mu.semte.ch/vocabularies/ext/formSubmissionStatus" },
+          "object": { "type": "uri", "value": "http://lblod.data.gift/concepts/9bd8d86d-bb10-4456-a84e-91e9507c374c" }
+        } 
+      ]
```
Comment (Riadabd) [inline scripts/send-submissions-to-prepare-for-export/run.sh:16]: Very small nitpick:
```suggestion
        }
      ]
```

I noticed there was extra spaces at the end of lines 15 and 16.

**Comment by claire-lovisa** [comment]
Closed because merged manually as a hotfix and then propagated to master and dev branches

**Pull request closed without merging**